File: src/main/java/com/zunza/buythedip/auth/oauth2/dto/KakaoOAuth2Response.java
package com.zunza.buythedip.auth.oauth2.dto;

import java.util.Map;

import com.zunza.buythedip.user.constant.OAuth2Provider;

import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
public class KakaoOAuth2Response implements OAuth2Response {
	private final Map<String, Object> attribute;

	@Override
	public OAuth2Provider getProvider() {
		return OAuth2Provider.KAKAO;
	}

	@Override
	public String getProviderId() {
		return attribute.get("id").toString();
	}

	@Override
	public String getEmail() {
		Map<String, Object> kakaoAccount = (Map<String, Object>) attribute.get("kakao_account");
		return kakaoAccount.get("email").toString();
	}
}

File: src/main/kotlin/com/zunza/buythedip/auth/oauth2/dto/KakaoOAuth2Response.kt
package com.zunza.buythedip.auth.oauth2.dto

import com.zunza.buythedip.user.constant.OAuth2Provider

class KakaoOAuth2Response(
    private val attribute: Map<String, Any>
) : OAuth2Response {

    override fun getProvider(): OAuth2Provider {
        return OAuth2Provider.KAKAO
    }

    override fun getProviderId(): String {
        return attribute["id"].toString()
    }

    override fun getEmail(): String {
        val kakaoAccount: Map<String, Any> = attribute["kakao_account"] as Map<String, Any>
        return kakaoAccount["email"].toString()
    }
}
